app/services/flow/flow_state_machine.rb

```diff
@@ -61,6 +61,16 @@ def register_campaign
       Funnel::DocAuth::RegisterCampaign.call(user_id, session[:ial2_with_no_sp_campaign])
     end
 
+    def analytics_user_override_hash
+      uuid = uuid_of_actual_user_if_user_is_anonymous
+      uuid ? {user_id: uuid} : {}
+    end
+
+    def uuid_of_actual_user_if_user_is_anonymous
+      return unless current_user && user_id_from_token
```

Comment:
Per prior comment, I don't think this would work as expected, since AFAIK either `current_user` or `user_id_from_token` would be truthy, but not both.

If the idea is that this would return a value only for a hybrid session user, should this instead be...

```suggestion
      return unless !current_user && user_id_from_token
```

or even...


```suggestion
      return unless user_id_from_token
```

Reply:
still making updates...